ext/search_kit/Civi/Search/Admin.php

```diff
@@ -366,6 +366,24 @@ public static function getJoins(array $allowedEntities):array {
             }
           }
         }
+        // Custom EntityRef joins
```

Comment:
@MegaphoneJon did you happen to notice what in the above code stops it from working for custom fields already? I would have thought their metadata would have triggered it to include them since they do declare a `fk_entity` like other fields.

Reply:
@colemanw I'm not sure which of places you mean by "above code", but:
* Line 329 references `fk_entity` but is in an `if ($bridge)` conditional.
* Lines 273-319 handle non-bridge joins, but only loops through `$references`. Since `$references` comes from  `$dao::getReferenceColumns()` it doesn't handle custom fields.
* `addImplicitFKFields()` works correctly (because of your work on #25927), but only `getJoins()` modifies `$joins`.  This leaves a scenario where you can hack `civicrm_saved_search.api_params` to retrieve the fields you want - but you can't select the joins in the SK UI.

Reply:
Ah, makes sense. Ok.
Thanks for adding a test.